src/profile-logic/sanitize.js

```diff
@@ -0,0 +1,224 @@
+import type { RemoveProfileInformation } from '../types/profile-derived';
+
+/**
+ * Take a processed profile with PII that user wants to be removed and remove the
+ * thread data depending on that PII status. Look at `RemoveProfileInformation`
+ * type definition if you want to learn what kind of information we are removing.
+ */
+export function sanitizePII(
+  profile: Profile,
+  PIIToBeRemoved: RemoveProfileInformation
+): Profile {
+  let urlCounter = 0;
+  const oldThreadIndexToNew: Map<ThreadIndex, ThreadIndex | null> = new Map();
+  let pages;
+  if (profile.pages) {
+    if (
+      PIIToBeRemoved.shouldRemoveNetworkUrls ||
+      PIIToBeRemoved.shouldRemoveAllUrls
+    ) {
+      pages = profile.pages.map(page =>
+        Object.assign({}, page, {
+          url: 'Page #' + urlCounter++,
+        })
+      );
+    } else {
+      pages = profile.pages;
+    }
+  } else {
+    pages = undefined;
+  }
+
+  const newProfile = {
+    ...profile,
+    meta: {
+      ...profile.meta,
+      extensions: PIIToBeRemoved.shouldRemoveExtensions
+        ? getEmptyExtensions()
+        : profile.meta.extensions,
+    },
+    pages: pages,
+    threads: profile.threads.reduce((acc, thread, threadIndex) => {
+      const newThread: Thread | null = sanitizeThreadPII(
+        thread,
+        threadIndex,
+        PIIToBeRemoved
+      );
+
+      if (newThread === null) {
+        oldThreadIndexToNew.set(threadIndex, null);
+        // Filtering out the current thread if it's null.
+        return acc;
+      }
+
+      // Adding the thread to the `threads` list.
+      oldThreadIndexToNew.set(threadIndex, acc.length);
+      return acc.concat(newThread);
+    }, []),
+    // Remove counters which belong to the removed counters.
+    // Also adjust other counters to point to the right thread.
+    counters: profile.counters
+      ? profile.counters.reduce((acc, counter) => {
+          const newThreadIndex = oldThreadIndexToNew.get(
+            counter.mainThreadIndex
+          );
+          if (!newThreadIndex) {
+            // Filtering out the current counter.
+            return acc;
+          }
+
+          counter.mainThreadIndex = newThreadIndex;
+          // Adding the current counter to the `counters` list.
+          return acc.concat(counter);
+        }, [])
+      : undefined,
+  };
+
+  return newProfile;
+}
+
+/**
+ * We want to protect users from unknowingly uploading sensitive data, however
+ * this gets in the way of engineers profiling on nightly. For a compromise
+ * between good data sanitization practices, and not dropping useful information,
+ * do not sanitize on nightly and custom builds.
+ */
+export function getShouldSanitizeByDefault(profile: Profile): boolean {
+  switch (profile.meta.updateChannel) {
+    case 'default': // Custom builds.
+    case 'nightly':
+    case 'nightly-try':
+      return false;
+    default:
+      return true;
+  }
+}
+/**
```

Comment:
While you are here, could you also add a new line before the comment please? (it just disturbed me, idk 😄 )